File: 250211/모이자/gather.py
n = int(input())
A = list(map(int, input().split()))

# Write your code here!
answer = float('inf')
for i in range(n):
    temp = 0
    for j in range(n):
        if i == j:
            continue
        temp += abs(i - j) * A[j]

    answer = min(answer, temp)

print(answer) 

File: 260416/모이자/gather.java
import java.io.*;
import java.util.*;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringTokenizer st;

    public static void main(String[] args) throws IOException {
        // Please write your code here.
        int n = Integer.valueOf(br.readLine());
        st = new StringTokenizer(br.readLine());
        int[] arr = new int[n];
        for(int i = 0; i < n; i++) {
            arr[i] = Integer.valueOf(st.nextToken());
        }

        int answer = Integer.MAX_VALUE;
        for(int i = 0; i < n; i++) {
            int temp = 0;
            int center = i;
            for(int j = i+1; j < n; j++) {
                temp += ((j - i) * arr[j]);
            }
            for(int j = i-1; j > -1; j--) {
                temp += ((i - j) * arr[j]);
            }

            answer = Math.min(answer, temp);
        }
        System.out.println(answer);
    }
}